src/opm/parser/eclipse/EclipseState/Schedule/Group/GuideRate.cpp

```diff
@@ -334,4 +322,28 @@ double GuideRate::get_grvalue_result(const GRValState& gr) const
         : std::max(gr.curr.value, 0.0);
 }
 
+
+void GuideRate::checkTimeToUpdate(const double sim_time, const size_t report_step)
+{
+    const auto& config = this->schedule[report_step].guide_rate();
+
+    if (!config.has_model()) {
+        this->time_to_update = false;
+        return;
+    }
+
+    // getting the last update time
+    double last_update_time = std::numeric_limits<double>::max();
+    for ([[maybe_unused]] const auto& [wgname, value] : this->values) {
+        const double update_time = value->curr.sim_time;
+//         if (value->curr.value > 0 && update_time < last_update_time) {
+        if (update_time < last_update_time) {
+            last_update_time = update_time;
+        }
+    }
```

Comment:
It is a good suggestion, will do.

Reply:
Sorry, there is some difficulty here. Because there might be situations that all the guide rates have zero values. In the final commit, 6552613b3827badb40e13611c589ba9347af00e4 , we are trying to discard all the zero valued guide rates when coming to update time checking.